src/allmydata/storage_client.py

```diff
@@ -1024,3 +1025,100 @@ def advise_corrupt_share(
             shnum,
             reason,
         ).addErrback(log.err, "Error from remote call to advise_corrupt_share")
+
+
+
+@attr.s
+class _FakeRemoteReference(object):
+    """
+    Emulate a Foolscap RemoteReference, calling a local object instead.
+    """
+    local_object = attr.ib(type=object)
+
+    def callRemote(self, action, *args, **kwargs):
+        return getattr(self.local_object, action)(*args, **kwargs)
```

Comment:
Nine other implementations of `IRemoteReference` in Tahoe (or implementations of `callRemote`, at least) in Tahoe and none of them is suitable to be used here.  Woo.

`foolscap.referenceable.LocalReferenceable` is closer than anything in Tahoe but since it wants to dispatch to `remote_` prefixed methods I guess it still isn't usable here.

Oh well.